pkg/operator/configobservation/apiserver/observe_tlssecurityprofile.go

```diff
@@ -0,0 +1,151 @@
+package apiserver
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/klog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+)
+
+func ObserveTLSSecurityProfile(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	var (
+		minTlSVersionPath = []string{"servingInfo", "minTLSVersion"}
+		cipherSuitesPath  = []string{"servingInfo", "cipherSuites"}
+	)
+
+	listers := genericListers.(configobservation.Listers)
+	errs := []error{}
+
+	prevObservedConfig := map[string]interface{}{}
+	currentMinTLSVersion, _, err := unstructured.NestedString(existingConfig, minTlSVersionPath...)
+	if err != nil {
+		return prevObservedConfig, append(errs, fmt.Errorf("failed to retrieve spec.servingInfo.minTLSVersion: %v", err))
+	}
+	if len(currentMinTLSVersion) > 0 {
+		if err = unstructured.SetNestedField(prevObservedConfig, currentMinTLSVersion, minTlSVersionPath...); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	currentCipherSuites, _, err := unstructured.NestedStringSlice(existingConfig, cipherSuitesPath...)
+	if err != nil {
+		return prevObservedConfig, append(errs, fmt.Errorf("failed to retrieve spec.servingInfo.cipherSuites: %v", err))
+	}
+	if len(currentCipherSuites) > 0 {
+		if err = unstructured.SetNestedStringSlice(prevObservedConfig, currentCipherSuites, cipherSuitesPath...); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	apiServer, err := listers.APIServerLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		klog.Warningf("apiserver.config.openshift.io/cluster: not found")
+		apiServer = &configv1.APIServer{}
+	} else if err != nil {
+		return existingConfig, append(errs, err)
+	}
+
+	observedConfig := map[string]interface{}{
+		"servingInfo": map[string]interface{}{},
+	}
+	observedMinTLSVersion, observedCipherSuites := GetSecurityProfileCiphers(apiServer.Spec.TLSSecurityProfile)
+	if err = unstructured.SetNestedField(observedConfig, observedMinTLSVersion, minTlSVersionPath...); err != nil {
+		errs = append(errs, err)
+	}
+	if err = unstructured.SetNestedStringSlice(observedConfig, observedCipherSuites, cipherSuitesPath...); err != nil {
+		errs = append(errs, err)
+	}
+
+	if string(observedMinTLSVersion) != currentMinTLSVersion {
+		recorder.Eventf("ObserveTLSSecurityProfile", "minTLSVersion changed to %s", observedMinTLSVersion)
```

Comment:
```suggestion
		recorder.Eventf("ObserveTLSSecurityProfile", "New minTLSVersion value observed: %s", observedMinTLSVersion)
```

Reply:
the "changed to" wording is used in images/scheduler/cors observations, similar is used in etcd observations, I'd stick with it.

Reply:
I'm not crazy about the wording since it hasn't "changed" until the observed config is successfully updated (but it is a nit) and the config observer controller [emits an event with the changes](https://github.com/openshift/library-go/blob/aa3ecef6350f28539223a61e55de596d56f582a4/pkg/operator/configobserver/config_observer_controller.go#L117).

Reply:
You can do a follow-up to this changing all the above mentioned observers having this wording :slightly_smiling_face: